vector/src/main/java/im/vector/app/features/home/room/detail/timeline/action/MessageActionsViewModel.kt

```diff
@@ -287,7 +290,24 @@ class MessageActionsViewModel @AssistedInject constructor(@Assisted
     private fun ArrayList<EventSharedAction>.addActionsForSendingState(timelineEvent: TimelineEvent) {
         // TODO is uploading attachment?
         if (canCancel(timelineEvent)) {
-            add(EventSharedAction.Cancel(timelineEvent.eventId))
+            add(EventSharedAction.Cancel(timelineEvent.eventId, false))
+        }
+    }
+
+    private fun ArrayList<EventSharedAction>.addActionsForSentNotSyncedState(timelineEvent: TimelineEvent,
+                                                                             actionPermissions: ActionPermissions,
+                                                                             messageContent: MessageContent?,
+                                                                             msgType: String?) {
+        if (timelineEvent.root.sendState == SendState.SENT) {
```

Comment:
This test is already done before calling this method, so we could remove it

Reply:
Root cause is complex and hard to catch on synapse side, two main scenario could cause that:
- It could be the downstream bug (https://github.com/matrix-org/synapse/issues/9624)
- Or a gappy sync scenario (two much events, and the synced event is in the gap, and in this case clients should have some mechanism to backfill if a sent event is not synced after following sync...)